holoviews/core/util.py

```diff
@@ -692,18 +704,69 @@ def isnumeric(val):
         return False
 
 
+def asarray(arraylike, strict=True):
+    """
+    Converts arraylike objects to NumPy ndarray types. Errors if
+    object is not arraylike and strict option is enabled.
+    """
+    if isinstance(arraylike, np.ndarray):
+        return arraylike
+    elif isinstance(arraylike, list):
+        return np.asarray(arraylike, dtype=object)
+    elif not isinstance(arraylike, np.ndarray) and isinstance(arraylike, arraylike_types):
+        return arraylike.values
+    elif hasattr(arraylike, '__array__'):
+        return np.asarray(arraylike)
+    elif strict:
+        raise ValueError('Could not convert %s type to array' % type(arraylike))
+    return arraylike
+
+
+nat_as_integer = np.datetime64('NAT').view('i8')
+
+def isnat(val):
+    """
+    Checks if the value is a NaT. Should only be called on datetimelike objects.
+    """
+    if isinstance(val, (np.datetime64, np.timedelta64)):
+        if numpy_version >= '1.13':
+            return np.isnat(val)
+        else:
+            return val.view('i8') == nat_as_integer
+    elif pd and val is pd.NaT:
+        return True
+    elif pd and isinstance(val, pandas_datetime_types+pandas_timedelta_types):
+        return pd.isna(val)
+    else:
+        return False
+
+
 def isfinite(val):
     """
     Helper function to determine if scalar or array value is finite extending
     np.isfinite with support for None, string, datetime types.
     """
+    if not np.isscalar(val):
+        val = asarray(val, strict=False)
+
     if val is None:
         return False
     elif isinstance(val, np.ndarray):
-        if val.dtype.kind in 'USMO':
+        if val.dtype.kind == 'M':
+            if pd:
+                return ~pd.isna(val)
+            elif numpy_version >= '1.13':
+                return ~np.isnat(val)
+            else:
+                return val.view('i8') != nat_as_integer
```

Comment:
And this logic can't be folded into isnat above?

Reply:
Done.

Reply:
Looks good; thanks.